### PR TITLE
Remove obsolete reference to old default local parallelism in Kafka source javadoc

### LINKED ISSUE #2663: [javadoc] default local parallelism for Kafka sources is ambiguous
Hi, first of all thanks for Hazelcast Jet. It is a really well-engineered product, kudos!

Currently, the official javadoc for the Kafka sources states two different default local parallelism values.

First time:

https://github.com/hazelcast/hazelcast-jet/blob/90d17d6be91cf4e7332059b47e5a9cadefd8ee3b/extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaSources.java#L65-L66

Second time further down below at the end:

https://github.com/hazelcast/hazelcast-jet/blob/90d17d6be91cf4e7332059b47e5a9cadefd8ee3b/extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaSources.java#L115-L116

I went to the source code to find out what is the actual default local parallelism being used, which is `4`:

https://github.com/hazelcast/hazelcast-jet/blob/90d17d6be91cf4e7332059b47e5a9cadefd8ee3b/extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaProcessors.java#L40

Following the example of the Kafka sink (see below), and for consistency, I suggest to drop the sentence mentioning the wrong default local parallelism of `2` in the first instance at the Kafka source documentation and only mention the default in the last paragraph as it is done for the sink documentation.

---

https://github.com/hazelcast/hazelcast-jet/blob/90d17d6be91cf4e7332059b47e5a9cadefd8ee3b/extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaSinks.java#L45-L99

### PR DESCRIPTION
**Pull request by hhromic**
Fixes #2663 

NO Breaking changes.

Checklist:
- [x] Labels and Milestone set

### CODE REVIEW

File: extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaSources.java [hunk 1]
```diff
@@ -62,10 +62,9 @@ private KafkaSources() {
      * The source creates a {@code KafkaConsumer} for each {@link Processor}
      * instance using the supplied {@code properties}. It assigns a subset of
      * Kafka partitions to each of them using manual partition assignment (it
-     * ignores the {@code group.id} property). Default local parallelism for
-     * this processor is 2 (or less if less CPUs are available). The Kafka's
-     * message timestamp will be used as a {@linkplain
-     * StreamSourceStage#withNativeTimestamps(long) native timestamp}.
+     * ignores the {@code group.id} property).
```
Comment (mtopolnik) [inline extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaSources.java:65]: Please reflow the paragraph, the rule is 72 chars per line (not counting initial indentation or `*`).

Comment (hhromic) [reply extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaSources.java]: FIxed this now. The following line is 74 characters for the sake of readability:

    will be used as a {@linkplain StreamSourceStage#withNativeTimestamps(long)

There are several lines with more than 72 characteres so I thought it is ok.
Let me know if this is still unacceptable.

Comment (mtopolnik) [reply extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaSources.java]: Yeah, ideally those would be improved as well :) But this is fine.

**Comment by mtopolnik** [comment]
verify

**Review by mtopolnik — APPROVED** [review]

**Pull request merged**